tools/sdk-generation-pipeline/packages/sdk-generation-lib/src/lib/ResultPublisher.ts

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public async sendSdkTaskResultToDB(pipelineBuildId: string, taskResults: TaskResult[]) {
+        if (pipelineBuildId === undefined) {
+            throw new Error('pipelineBuildId is empty!');
+        }
+
+        const taskResultDao: TaskResultDao = new TaskResultDaoImpl(this.connection);
+        for (const taskResult of taskResults) {
+            await taskResultDao.put(pipelineBuildId, taskResult);
+        }
+    }
+
+    public async sendSdkGenerationToDB(cg: CodeGeneration) {
+        const codeGenerationDao: CodeGenerationDao = new CodeGenerationDaoImpl(this.connection);
+
+        await codeGenerationDao.submitCodeGeneration(cg);
+    }
+}
+
+export type BlobBasicContext = {
+    pipelineBuildId: string;
+    sdkGenerationName: string;
+    azureStorageBlobSasUrl: string;
+    azureBlobContainerName: string;
+};
+
+export class ResultBlobPublisher {
+    private pipelineBuildId: string;
+    private sdkGenerationName: string;
+    private azureBlobClient: AzureBlobClient;
+
+    constructor(conetxt: BlobBasicContext) {
+        this.pipelineBuildId = conetxt.pipelineBuildId;
+        this.sdkGenerationName = conetxt.sdkGenerationName;
+        this.azureBlobClient = new AzureBlobClient(conetxt.azureStorageBlobSasUrl, conetxt.azureBlobContainerName);
+    }
+
+    public async uploadLogsAndResult(logsAndResultPath: string, taskNameInput: AzureSDKTaskName) {
+        let taskName: string = taskNameInput;
+
+        if (!taskNameInput) {
+            logger.info(`taskName is undefined or null, set full`);
+            taskName = 'full';
+        }
+
+        const logsAndResultPathArray = JSON.parse(logsAndResultPath);
+        for (const file of logsAndResultPathArray) {
+            if (fs.existsSync(file)) {
+                let blobName: string = file.includes('.json')
+                    ? `${this.pipelineBuildId}/logs/${this.sdkGenerationName}-${taskName}-result.json`
+                    : `${this.pipelineBuildId}/logs/${this.sdkGenerationName}-${taskName}.log`;
+                await this.azureBlobClient.publishBlob(file, blobName);
+                logger.info(`Publish ${file} Success !!!`);
+            } else {
+                logger.error(`Publish result failed !, ${file} is missed`);
+            }
+        }
+    }
+}
+
+export class ResultEventhubPublisher {
+    private producer: EventHubProducer;
+
+    constructor(eventHubConnectionString: string) {
+        this.producer = new EventHubProducer(eventHubConnectionString);
+    }
+
+    public async publishEvent(event: PipelineRunEvent, partitionKey?: string): Promise<void> {
+        try {
+            await this.producer.send([JSON.stringify(event)], partitionKey);
+        } catch (e) {
+            await this.producer.close();
+            logger.error('Failed to send pipeline result:', JSON.stringify(event), e);
+            throw e;
+        }
+    }
+
+    public generateTotalResult(taskResults: TaskResult[], pipelineBuildId: string): TaskResult {
```

Comment:
This function is not belong to class 'ResultEventhubPublisher'. It is a function to create result and wrap it as event, then call publishEvent. 

Remove this method out of the class, and put it in someplace as a util

Reply:
yes, you are right, generateTotalResult is only related to the task result, I will put it in taskResult.ts